presto-main/src/test/java/io/prestosql/sql/planner/assertions/SemiJoinMatcher.java

```diff
@@ -64,6 +82,38 @@ public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session ses
             return NO_MATCH;
         }
 
+        if (dynamicFilteringEnabled) {
+            if (!semiJoinNode.getDynamicFilterId().isPresent()) {
+                return NO_MATCH;
+            }
+            String dynamicFilterId = semiJoinNode.getDynamicFilterId().get();
```

Comment:
Could you follow same pattern as for `io.prestosql.sql.planner.assertions.DynamicFilterMatcher`
It might be that filtered symbol is aliased by by projection, so you need up-to-date `SymbolAliases` (children are matched first) instaed of searching for filter node.